src/components/WalletConnect/CreateTokenRequest.js

```diff
@@ -0,0 +1,209 @@
+import checkIcon from '../../assets/images/icCheckBig.png';
+
+const condRenderData = (
+  attribute,
+  title,
+  separator,
+  formatter = (val) => val,
+) => {
+  if (attribute != null) {
+    return (
+      <>
+        { separator && <View style={commonStyles.cardSeparator} /> }
+        <View>
+          <Text style={styles.property}>{ title }</Text>
+          <Text style={[styles.value, styles.bold]}>
+            { formatter(attribute) }
+          </Text>
+        </View>
+      </>
+    );
+  }
+
+  return null;
+};
+
+function renderBoolean(bool) {
+  return bool ? t`Yes` : t`No`;
+}
+
+export const CreateTokenRequestData = ({ data }) => (
+  <View style={[commonStyles.card, commonStyles.cardSplit]}>
+    <View style={commonStyles.cardSplitContent}>
+      { condRenderData(data.name, t`Token Name`, false) }
+      { condRenderData(data.symbol, t`Token Symbol`, true) }
+      { condRenderData(data.amount, t`Token Amount`, true, numberUtils.prettyValue) }
+      { condRenderData(data.changeAddress, t`Address to send change ${DEFAULT_TOKEN.uid}`, true) }
+      { condRenderData(data.mintAuthorityAddress, t`Address to send the mint authority`, true) }
+      { condRenderData(data.meltAuthorityAddress, t`Address to send the melt authority`, true) }
+      { condRenderData(data.allowExternalMintAuthorityAddress, t`Allow external mint authority addresses?`, true, renderBoolean) }
+      { condRenderData(data.allowExternalMeltAuthorityAddress, t`Allow external melt authority addresses?`, true, renderBoolean) }
+      { condRenderData(data.createMint, t`Create mint authority?`, true, renderBoolean) }
+      { condRenderData(data.createMelt, t`Create melt authority?`, true, renderBoolean) }
+      { condRenderData(data.data, t`Token data`, true, (tokenData) => tokenData.split('\n')) }
+    </View>
+  </View>
+);
+
+export const CreateTokenRequest = ({ createTokenRequest }) => {
+  const { dapp, data } = createTokenRequest;
+  const { message, address } = data;
+  const { retrying, status } = useSelector((state) => state.walletConnect.createToken);
+  const dispatch = useDispatch();
+  const navigation = useNavigation();
+
+  const onAcceptCreateTokenRequest = () => {
+    const acceptedNc = { address, message };
+
+    dispatch(walletConnectAccept(acceptedNc));
+  };
+
+  const onDeclineTransaction = () => {
+    dispatch(walletConnectReject());
+    navigation.goBack();
+  };
+
+  const isTxReady = status === WALLETCONNECT_CREATE_TOKEN_STATUS.READY;
+  const isTxProcessing = status === WALLETCONNECT_CREATE_TOKEN_STATUS.LOADING;
+  const isTxSuccessful = status === WALLETCONNECT_CREATE_TOKEN_STATUS.SUCCESSFUL;
+  const isTxFailed = status === WALLETCONNECT_CREATE_TOKEN_STATUS.FAILED;
+
+  const onFeedbackModalDismiss = () => {
+    dispatch(setCreateTokenStatusReady());
+    dispatch(createTokenRetryDismiss());
+    navigation.goBack();
+  };
+
+  const onNavigateToDashboard = () => {
+    dispatch(setCreateTokenStatusReady());
```

Comment:
Why not call `createTokenRetryDismiss`?

Reply:
Refactored it to set ready on component unmount